terraform/create.go

```diff
@@ -103,11 +230,18 @@ func (t *Terraform) runCommand(args ...string) error {
 	}
 
 	rdr := io.MultiReader(stdout, stderr)
-	scanner := bufio.NewScanner(rdr)
-	for scanner.Scan() {
-		mlog.Info(scanner.Text())
+	if dst != nil {
+		_, err = io.Copy(dst, rdr)
+		if err != nil {
+			return err
+		}
+	} else {
+		scanner := bufio.NewScanner(rdr)
+		for scanner.Scan() {
+			mlog.Info(scanner.Text())
+		}
+		// No need to check for scanner.Error as cmd.Wait() already does that.
```

Comment:
If I understand this correctly, if we provide an `io.Writer` we read output of `cmd` to `dst` but if not; we are logging it. I think we shouldn't log here. This responsibility should be in the caller.

Reply:
Maybe, but then all the callers will have loops consisting of `mlog.Info`. That's a lot of repetition. Moreover, now with this `dst` variable, you would need an additional check to see whether `dst` was passed or not, and if not, then log it. So `dst` would need to be checked in all the callers again. Quite a lot of additional code for not much benefit IMO.